index.bs

```diff
@@ -13202,6 +13394,69 @@ The characteristics of a namespace object are described in [[#namespace-object]]
 </div>
 
 
+<h3 id="es-modules">Modules</h3>
+
+Modules are reified as entries in the [=module map=] as [=Synthetic Module
+Records=], allowing them to be imported as ECMAScript modules.
+
+<div algorithm>
+  To <dfn>create a synthetic module record</dfn> for a [=module=] |module| in a
+  [=Realm=] |realm|, perform the following steps:
+
+    1.  Let |exports| be « ».
+    1.  [=list/For each=] [=module member=] |member| of |module|:
+        1.  Assert: |member| is not annotated with either the [{{Exposed}}]
+                    or [{{SecureContext}}] extended attributes.
+        1.  Append the [=identifier=] of |member| to |exports|.
+    1.  Let |moduleRecord| be ! [=CreateSyntheticModule=](|exports|, the following steps, |realm|, undefined), with the following steps:
+        1. [=list/For each=] [=module member=] |member| of |module|:
+            1.  Let |id| be |member|'s [=identifier=].
+            1.  If |member| is an [=interface=]:
+                1.  Let |interfaceObject| be the result of [=create an interface
+                    object|creating an interface object=] for |member| with |id|
+                    in |realm|.
+                1.  Perform ! [=SetSyntheticModuleExport=](|moduleRecord|, |id|,
+                    |interfaceObject|).
+            1.  Otherwise, if |member| is an [=operation=]:
+                1.  Let |method| be the result of [=creating an operation function=]
+                    given |member|, |module|, and |realm|.
+                1.  Perform ! [=SetSyntheticModuleExport=](|moduleRecord|, |id|,
+                    |method|).
+            1.  Otherwise:
+                1.  Assert: |member| is an [=attribute=].
+                1.  Assert: The module evaluation steps will set each attribute.
+        1.  Run the [=module evaluation steps=] for the [=module/specifier=] of
+            |module|, if any are present.
+    1.  Return |moduleRecord|.
+</div>
+
+<div algorithm>
+  To <dfn>initialize the module map</dfn> of a [=Realm=] |realm|:
+
+    1.  Let |map| be the [=module map=] associated with |realm|'s [=relevant settings object=].
+    1.  For every [=module=] |module| that is [=exposed=] in |realm|,
+        1.  Let |name| be the [=module/specifier=] of |module|.
+        1.  Let |record| be the result of [=create a synthetic module record=]
+            for |module| in |realm|.
+        1.  [=map/Set=] |map|[|name|] to |record|.
```

Comment:
This line doesn't really make sense. The module map is keyed by URLs, whereas specifiers are strings (even if we concatenate a prefix on). Should we make built-in module specifiers be URLs? Or make the map somehow heterogeneous, with special handling in HTML's module specifier resolution algorithm? Or something else?

Reply:
https://github.com/WICG/import-maps/tree/master/reference-implementation/lib has been working off the assumption that the module map keys would be `import:x` for built-in modules with specifier `x`. https://github.com/WICG/import-maps/blob/master/reference-implementation/lib/parser.js#L110